Refactor Club Fantastic cmake integration to avoid copying unwanted song packs into ITGmania.app when building on macOS

sukibaby opened the pull request:
The current code copies the entire contents of the /Songs folder **within ITGmania.app itself**.  So, if you have 20GB of packs when you compile the game, you end up with a 20GB sized executable.

<img width="1214" height="801" alt="Screenshot 2025-11-03 at 7 18 06 PM" src="https://github.com/user-attachments/assets/294921dc-5d97-494e-a929-30dcc59f0cd2" />

It's hard to imagine this was intended to do any more than maybe bundle the three default StepMania 5 songs, so I've removed it from the CMake.

sukibaby committed:
Refactor Club Fantastic integration in CMake

The current code has two problems. First, on macOS, it copies the entire contents of the /Songs folder within ITGmania.app itself. So, if you have 20GB of packs when you compile the game, you end up with a 20GB sized executable. Second, you can end up with two copies of Club Fantastic songs on macOS (one in the Songs folder and one in the .app bundle). This refactors Club Fantastic CMake integration to the top level so that all platforms can be built as expected and unwanted or duplicate song packs will not end up in the macOS builds.

teejusb commented:
Hmm this *is* intended for the "with songs" binary to include Club Fantastic. I presume one can use additional folders in preferences to prevent bundling excess packs?

sukibaby commented:
True, will see if i can make that cmake a little nicer though.

sukibaby commented:
Refactoring to allow Club Fantastic songs to be built-in as expected.

sukibaby committed:
Add note about macOS bundle resource copying

Clearly indicate to readers what is happening here, since this behavior is not clearly defined.
